pkg/server/commands/write.go

```diff
@@ -65,6 +65,9 @@ func (c *WriteCommand) validateWriteRequest(ctx context.Context, req *openfgapb.
 
 		authModel, err := c.datastore.ReadAuthorizationModel(ctx, store, modelID)
 		if err != nil {
+			if errors.Is(err, storage.ErrNotFound) {
```

Comment:
Though this addition is a good one, the overall logic contained herein seems a bit faulty.

It appears as though we attempt to read (and therefore validate the existence of) the authorization model `if len(writes) > 0`. But what if you don't specify any writes but you do specify some deletes? Should we not also verify the authorization model exists in that case?

It seems this code should change to:

```go
authModel, err := c.datastore.ReadAuthorizationModel(ctx, store, modelID)
if err != nil {
    if errors.Is(err, storage.ErrNotFound) {
        return serverErrors.AuthorizationModelNotFound(modelID)
    }
}

if len(writes) > 0 {
    // do existing logic
}

for _, tk := range deletes {
    // do existing delete validation
}
```

Reply:
Please add a test for this case as well ☝️

Reply:
For writes, we need to do read authorization model for type validation purpose.  For delete, we do not care about auth model as we do not need to do type validation.